### PR TITLE
docs: add General Availability banner to README

### PR DESCRIPTION
**Pull request by tswast**
As of 1.0.0 release, BigQuery DataFrames is available as a [Generally Available](https://github.com/googleapis/google-cloud-python/blob/main/README.rst#general-availability) Google Cloud product.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Comment by tswast** [comment]
Wait until 4/4 to merge.

**Review by TrevorBergeron — APPROVED** [review]

**Pull request merged**
